code/_onclick/xeno.dm

```diff
@@ -2,10 +2,8 @@
 	if(lying_angle)
 		return FALSE
 
-	if(xeno_caste)
-		changeNext_move(xeno_caste.attack_delay)
-	else
-		changeNext_move(CLICK_CD_MELEE)
+	if(!isturf(A)) //We don't care about turfs; they don't trigger our melee click cooldown
+		changeNext_move(xeno_caste ? xeno_caste.attack_delay : CLICK_CD_MELEE)
```

Comment:
cant this trigger on walls ? but i think xeno walls arent turfs right?

Reply:
I can differentiate for xeno wall turfs if deemed necessary

Reply:
resin walls are closed turfs that's why they block tad, why would they need to right-clicked anyway tho?

Reply:
Changenext move should trigger on turf clicks make this isopenturf maybe